skyportal/facility_apis/ztf.py

```diff
@@ -206,6 +363,123 @@ def delete(request):
 
         DBSession().add(transaction)
 
+    @staticmethod
+    def get(request):
+
+        """Get a forced photometry request result from ZTF.
+
+        Parameters
+        ----------
+        request : skyportal.models.FollowupRequest
+            The request to add to the queue and the SkyPortal database.
+        """
+
+        from ..models import (
+            DBSession,
+            FollowupRequest,
+            FacilityTransaction,
+            Allocation,
+            Instrument,
+        )
+
+        Session = scoped_session(
+            sessionmaker(bind=DBSession.session_factory.kw["bind"])
+        )
+        session = Session()
+
+        req = (
+            session.query(FollowupRequest)
+            .filter(FollowupRequest.id == request.id)
+            .one()
+        )
+
+        instrument = (
+            Instrument.query_records_accessible_by(req.requester)
+            .join(Allocation)
+            .join(FollowupRequest)
+            .filter(FollowupRequest.id == request.id)
+            .first()
+        )
+
+        altdata = request.allocation.altdata
+
+        if not altdata:
+            raise ValueError('Missing allocation information.')
+
+        keys = ['ra', 'dec', 'jdstart', 'jdend']
+
+        content = req.transactions[-1].response["content"]
+        df_request = pd.read_html(content)[0]
+        if df_request.shape[0] == 0:
+            raise ValueError('Missing response from forced photometry service.')
+        df_request.columns = df_request.columns.str.lower()
+        if not set(keys).issubset(df_request.columns):
+            raise ValueError("RA, Dec, jdstart, and jdend required in response.")
+        df_request = df_request.iloc[0]
+        df_request = df_request.replace({np.nan: None})
+
+        requestgroup = {
+            "email": altdata["ipac_email"],
+            "userpass": altdata["ipac_userpass"],
+            "option": "All recent jobs",
+            "action": "Query Database",
+        }
+        params = urllib.parse.urlencode(requestgroup)
+        url = f"{ZTF_FORCED_URL}/cgi-bin/getForcedPhotometryRequests.cgi?{params}"
+
+        r = requests.get(
+            url,
+            auth=HTTPBasicAuth(
+                altdata['ipac_http_user'], altdata['ipac_http_password']
+            ),
+        )
+        r.raise_for_status()
```

Comment:
Uncaught exception.

Reply:
I feel like handling the return codes below is the correct thing here as we do elsewhere in the app right?

Reply:
I'm not sure. Would `raise_for_status` raise an exception when it is called on `r` that failed?

Reply:
yeah thats true. https://stackoverflow.com/questions/61463224/when-to-use-raise-for-status-vs-status-code-testing. So maybe we just let the if statement catch it then?

Reply:
Yes. I think the right thing to do is one of:
- if in a handler function, just check the status code of `r` and `self.error` if needed.
- if in a helper function, you can use `raise_for_status` but make sure to catch that in the handler function.

Reply:
yes i went through and nixed them.